From a2575c3c96be502e4af420f50e6b816569330531 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sun, 22 Nov 2015 17:09:05 +0300
Subject: [PATCH 27/27] The operator new declaration has changed slightly in
 C++11.

It has to do with the fact that the throwing exception specification is deprecated in C++11,
and the introduction of the nothrow declaration:
throw (std::bad_alloc) has been omitted
throw() is replaced with nothrow
---
 src/common/classes/alloc.cpp | 8 ++++----
 src/common/classes/alloc.h   | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/common/classes/alloc.cpp b/src/common/classes/alloc.cpp
index 257030bccb..d6036ec2b2 100644
--- a/src/common/classes/alloc.cpp
+++ b/src/common/classes/alloc.cpp
@@ -2124,20 +2124,20 @@ void AutoStorage::ProbeStack() const
 
 } // namespace Firebird
 
-void* operator new(size_t s) THROW_BAD_ALLOC
+void* operator new(size_t s)
 {
 	return Firebird::MemoryPool::globalAlloc(s);
 }
-void* operator new[](size_t s) THROW_BAD_ALLOC
+void* operator new[](size_t s)
 {
 	return Firebird::MemoryPool::globalAlloc(s);
 }
 
-void operator delete(void* mem) throw()
+void operator delete(void* mem) noexcept
 {
 	Firebird::MemoryPool::globalFree(mem);
 }
-void operator delete[](void* mem) throw()
+void operator delete[](void* mem) noexcept
 {
 	Firebird::MemoryPool::globalFree(mem);
 }
diff --git a/src/common/classes/alloc.h b/src/common/classes/alloc.h
index 93ffe370cc..0c7e38f24d 100644
--- a/src/common/classes/alloc.h
+++ b/src/common/classes/alloc.h
@@ -499,11 +499,11 @@ using Firebird::MemoryPool;
 inline static MemoryPool* getDefaultMemoryPool() { return Firebird::MemoryPool::processMemoryPool; }
 
 // Global versions of operators new and delete
-void* operator new(size_t s) THROW_BAD_ALLOC;
-void* operator new[](size_t s) THROW_BAD_ALLOC;
+void* operator new(size_t s);
+void* operator new[](size_t s);
 
-void operator delete(void* mem) throw();
-void operator delete[](void* mem) throw();
+void operator delete(void* mem) noexcept;
+void operator delete[](void* mem) noexcept;
 
 #ifdef DEBUG_GDS_ALLOC
 inline void* operator new(size_t s, Firebird::MemoryPool& pool, const char* file, int line)
-- 
2.13.0

